modules/data.atmosphere/R/tdm_temporal_downscale.R

```diff
@@ -0,0 +1,655 @@
+##' Met downscaling functions that predict the downscaled meteorology 
+# -----------------------------------
+# Description
+# -----------------------------------
+##' @title temporal_downscale
+##' @family tdm - Temporally Downscale Meteorology
+##' @author Christy Rollinson, James Simkins
+##' @description This script contains the functions that do the heavy lifting in the final
+##'              script of the tdm workflow titled tdm_predict_subdialy_met.R. It uses a linear
+##'              regression approach by generating the hourly values from the daily means of
+##'              the file the user selects to downscale based on the hourly models and betas
+##'              generated by tdm_generate_subdaily_models.R. 
+# -----------------------------------
+# Parameters
+# -----------------------------------
+##' @param dat.mod - data to be predicted at the time step of the training data
+##' @param n.ens - number of hourly ensemble members to generate
+##' @param path.model - path to where the training model & betas is stored
+##' @param lags.init - a data frame of initialization paramters to match the data in dat.mod
+##' @param dat.train - the training data used to fit the model; needed for night/day in 
+##'                    surface_downwelling_shortwave_flux_in_air
+# -----------------------------------
+#----------------------------------------------------------------------
+# Begin Script
+#----------------------------------------------------------------------
+
+# Set progress bar
+pb.index <- 1
+pb <- txtProgressBar(min = 1, max = 8, style = 3)
+
+# Begin Function
+predict.subdaily <- function(dat.mod, n.ens, path.model, lags.list = NULL, 
+    lags.init = NULL, dat.train) {
+    
+    # -------------------------------- Models each variable separately at
+    # the moment rather than a generalized eqauiton that could potentially
+    # be parallilizeable -------------------------------- # 2.1
+    # air_temperature (air temperature) # 2.2 precipitation_flux
+    # (preciptiation, water equivalent) # 2.3
+    # surface_downwelling_shortwave_flux_in_air (downwelling shortwave
+    # radiation) # 2.4 surface_downwelling_longwave_flux_in_air
+    # (downwelling longwave radiation) # 2.5 air_pressure (surface
+    # air_pressureure) # 2.6 specific_humidity (specific humidity) # 2.7
+    # wind_speed (surface wind_speed speed)
+    # --------------------------------# Load libraries library(ncdf4)
+    # library(mgcv) library(MASS) library(ggplot2)
+    
+    setTxtProgressBar(pb, pb.index)
+    
+    # Figure out if we need to extract the approrpiate
+    if (is.null(lags.init)) {
+        lags.init <- lags.list[[unique(dat.mod$ens.day)]]
+    }
+    
+    # Set up the ensemble members in a list so the uncertainty can be
+    # propogated
+    dat.sim <- list()
+    
+    # ----- Modeling surface_downwelling_shortwave_flux_in_air 
+    {
```

Comment:
I don't think there would be any effect of removing them, so they are superfluous and comments would be sufficient.